src/Access/MemoryAccessStorage.h

```diff
@@ -36,6 +36,8 @@ class MemoryAccessStorage : public IAccessStorage
     bool isBackupAllowed() const override { return backup_allowed; }
     void restoreFromBackup(RestorerFromBackup & restorer) override;
 
+    bool updateNoLock(const UUID & id, const UpdateFunc & update_func, bool throw_if_not_exists) TSA_NO_THREAD_SAFETY_ANALYSIS;
```

Comment:
usually I do not like using friend classes because this makes code readability worse (just in my opinion)
but your explanation is clear and I agree and used `LDAPAccessStorage` as friend class instead of making `updateNoLock` public. Done.